### ISSUE TITLE
Einsy Rambo - Reprap Discount Full Graphics Garbled

### ISSUE THREAD

**Issue by Sloppsta** [MarlinFirmware/Marlin#10828]
### Description
I tried adding
```cpp
#define ST7920_DELAY_1 DELAY_NS(0)
#define ST7920_DELAY_2 DELAY_NS(4)
#define ST7920_DELAY_3 DELAY_NS(0)
```
to the ultralcd...st2930.h and also in the config.h but I keep getting a garbled screen. I have tried mutiple values and none seem to help. Suggestions?

### Steps to Reproduce

1. hook up einsy rambo and reprap full graphics display
2. turn on.

**Expected behavior:** [What you expect to happen]
Screen not garbled
**Actual behavior:** [What actually happens]
Screen Garbled
[Nicks Marlin 1.1.9 v1 Einsy Rambo June 2018.zip](https://github.com/MarlinFirmware/Marlin/files/2032483/Nicks.Marlin.1.1.9.v1.Einsy.Rambo.June.2018.zip)

#### Additional Information

* Include a ZIP file containing your `Configuration.h` and `Configuration_adv.h` files.
* Provide pictures or links to videos that clearly demonstrate the issue.
* See [How Can I Contribute](https://github.com/MarlinFirmware/Marlin/blob/1.1.x/.github/contributing.md#how-can-i-contribute) for additional guidelines.

**Comment by ejtagle** [comment]
Try higher values. 4ns (nanoseconds is absolutely nothing. Try 200... 500 ... 700nanoseconds as delay
DELAY_NS() does not measure in NOPs (as done previously). It measures time in nanoseconds.
A single NOP takes 62.5nanoseconds on AVR

**Comment by Sloppsta** [comment]
Thanks for the quick reply! I had no idea what a nope even was :P in the board definitions they are only adjusting the middle value (delay 2) should all three be adjusted or just keep trying to up delay 2? Thanks!

**Comment by ejtagle** [comment]
Try setting them all to 900, then reducing one by one. Those delays start adding quite a bit of time to the display handling, so the less the delays, the better

**Comment by Sloppsta** [comment]
I've worked my way up to
```cpp
        #define SDSUPPORT
	#define ST7920_DELAY_1 DELAY_NS(100000)
	#define ST7920_DELAY_2 DELAY_NS(100000)
	#define ST7920_DELAY_3 DELAY_NS(100000)
```

With no notable improvement. Any suggestions?

**Comment by ejtagle** [comment]
in such case the problems are not the delays themselves... We will have to wait until someone with this configuration is able to help ...

**Comment by amoose136** [comment]
I have the EinsyRambo and a RepRapDiscount full graphics display but have not yet hooked the two together. I'll report back if I encounter any problems/solutions to problems.

**Comment by larsnordstrom** [comment]
Enable #define DOGLCD

**Comment by thinkyhead** [comment]
Be sure to check that the keyed plugs aren't backwards. Sometimes this happens on boards, sometimes on displays. On the Einsy the key slots should be facing inwards.

**Comment by Sloppsta** [comment]
Thank you all for replying. @thinkyhead , I am a particular fan of your work!

I read somewhere else that doglcd is enabled automatically, so I didn't think I needed to do that.

Strangest thing though,after trying to set my delays to 900ns each as @ejtagle suggested, I decided to call it a miniRambo, and copied over the einsy pins to pins_minirambo- it did not work. When I flashed it back to einsy with the same config - it was working perfectly! All subsequent flashes have worked great too! I have no idea what happened but I am glad it's working great!


@thinkyhead, the keys were backwards - I discovered that early on.

Thank you all for your input!
